### PR TITLE
[codex] Validate plugin skill base names

### PR DESCRIPTION
**Pull request by xl-openai**
## Summary

- Validate skill base name length before plugin namespacing.
- Bound the composed `plugin:skill` qualified name to 128 characters.
- Keep plugin skill runtime names in the existing `plugin:skill` form.
- Add regression tests for the max qualified-name boundary and rejection path.

## Root Cause

Plugin skills are represented as `plugin_name:skill_name`, but the loader previously applied the 64-character skill name limit after adding the plugin namespace. Moving that check to the base name fixes valid plugin skills with longer namespaces, and the separate 128-character qualified-name limit keeps model-visible skill names bounded.

## Validation

- `just fmt`
- `just test -p codex-core-skills plugin_skill_name_length_limit`
- `git diff --check`

### CODE REVIEW

**Review by etraut-openai — APPROVED** [review]

**Comment by xl-openai** [comment]
@codex review again

File: codex-rs/core-skills/src/loader.rs [hunk 1]
```diff
@@ -660,7 +661,8 @@ async fn parse_skill_file(
         policy,
     } = load_skill_metadata(fs, path, plugin_root).await;
 
-    validate_len(&name, MAX_NAME_LEN, "name")?;
+    validate_len(&base_name, MAX_NAME_LEN, "name")?;
+    validate_len(&name, MAX_QUALIFIED_NAME_LEN, "qualified name")?;
```
[the remark this answers is not in the document]
Comment (xl-openai) [reply codex-rs/core-skills/src/loader.rs:665]: that's OK
